lang/java/protobuf/src/main/java/org/apache/avro/protobuf/ProtobufData.java

```diff
@@ -113,17 +114,17 @@ protected Object getField(Object record, String name, int pos, Object state) {
     }
   }
 
-  private final Map<Descriptor, FieldDescriptor[]> fieldCache = new ConcurrentHashMap<>();
+  private final Map<DescriptorProto, FieldDescriptor[]> fieldCache = new ConcurrentHashMap<>();
 
   @Override
   protected Object getRecordState(Object r, Schema s) {
     Descriptor d = ((MessageOrBuilder) r).getDescriptorForType();
-    FieldDescriptor[] fields = fieldCache.get(d);
+    FieldDescriptor[] fields = fieldCache.get(d.toProto());
     if (fields == null) { // cache miss
       fields = new FieldDescriptor[s.getFields().size()];
       for (Field f : s.getFields())
         fields[f.pos()] = d.findFieldByName(f.name());
-      fieldCache.put(d, fields); // update cache
+      fieldCache.put(d.toProto(), fields); // update cache
```

Comment:
Does it make sense to cache the result of `d.toProto()` and re-use it instead of calling it twice in the same method ?

Reply:
This is probably a good idea, internally, toProto() just does "return this.proto" but that is an implementation that might change.